### PR TITLE
Report HTTP signature parsing error to client 

### LINKED ISSUE #2890: Expose HttpSignatureVerifier::parse() failure details in HTTP response, and use them in *SignatureAuthnIdentity
The signature parser will currently log as TRACE failure details of signature verification, such as whether particular headers are missing etc, but won't make them available to the caller.

We should expose them, and return them in the HTTP response, to help guide users (or members) trying to sign payloads but hitting formatting issues.

### PR DESCRIPTION
**Pull request by jumaffre**
Resolves #2890

### CODE REVIEW

File: src/http/authentication/sig_auth.h [hunk 1]
```diff
@@ -72,37 +72,41 @@ namespace ccf
...
+      SignedReq signed_request;
+
+      try
+      {
+        signed_request = parse_signed_request(ctx);
+      }
+      catch (const std::exception& e)
+      {
+        error_reason = e.what();
+        return nullptr;
```
Comment (wintersteiger) [inline src/http/authentication/sig_auth.h:84]: Perhaps worth keeping the `std::optional` instead of `nullptr`s?

**Review by wintersteiger — APPROVED** [review]

File: src/http/authentication/sig_auth.h (hunk 1 above)
Comment (jumaffre) [reply src/http/authentication/sig_auth.h:84]: As discussed, the function already returned a `nullptr` on error (the default view for the GitHub diff is confusing) so I'll leave this as it is.

**Comment by jumaffre** [comment]
There seems to be a regression with `tpcc_sgx_cft`. Investigating.

**Comment by jumaffre** [comment]
> There seems to be a regression with `tpcc_sgx_cft`. Investigating.

The regression was due to the fact what we threw an exception on every unsigned request for HTTP endpoints that require a signature. This isn't the case anymore as we return a `std::nullopt` for unsigned requests (i.e. no `Authorization` header).

File: src/http/http_sig.h [hunk 2]
```diff
@@ -308,34 +307,31 @@ namespace http
         }
         else
         {
-          LOG_FAIL_FMT("Authorization parameter {} does not contain \"=\"", p);
-          return std::nullopt;
+          throw std::logic_error(fmt::format(
+            "authorization parameter '{}' does not contain \"=\"", p));
```
Comment (eddyashton) [inline src/http/http_sig.h:311]:
```suggestion
            "Authorization parameter '{}' does not contain \"=\"", p));
```

**Review by eddyashton — APPROVED** [review]

**Pull request merged**